src/state/session/index.tsx

```diff
@@ -406,34 +410,26 @@ export function Provider({children}: React.PropsWithChildren<{}>) {
         }
 
         // Intentionally not awaited to unblock the UI:
-        resumeSessionWithFreshAccount()
-          .then(freshAccount => {
-            if (JSON.stringify(account) !== JSON.stringify(freshAccount)) {
-              logger.info(
-                `session: reuse of previous session returned a fresh account, upserting`,
-              )
-              upsertAccount(freshAccount)
-            }
-          })
-          .catch(e => {
-            /*
-             * Note: `agent.persistSession` is also called when this fails, and
-             * we handle that failure via `createPersistSessionHandler`
-             */
-            logger.info(`session: resumeSessionWithFreshAccount failed`, {
-              message: e,
-            })
-
-            __globalAgent = PUBLIC_BSKY_AGENT
+        resumeSessionWithFreshAccount().catch(e => {
```

Comment:
The `persistSession` callback handles calling `upsertAccount`, we were calling it redundantly here.